### PR TITLE
[#LIEF-3326] Log routific API requests and responses

### PR DESCRIPTION
**Pull request by edwardmccaughan**

### CODE REVIEW

**Comment by stemps** [comment]
It would be nice if the new config options would be documented in the readme

**Comment by stemps** [comment]
... and if they would have explicit defaults (and comments) in the code. 

someting like in https://github.com/plataformatec/devise/blob/master/lib/devise.rb

File: lib/routific.rb [hunk 1]
```diff
@@ -62,11 +64,36 @@ def getRoute
...
 
+    # whether the result of requests should be logged
+    attr_reader :log_requests
+    @log_requests = false
+
+    # whether errors in the Api request and response parsing should throw exceptions or fail silently
+    attr_reader :raise_on_exception
+    @raise_on_exception = false
+
+    # the ruby Logger to be used to log errors, request logs etc
```
Comment (edwardmccaughan) [inline lib/routific.rb]: @stemps I tried making logger an attr_reader so it was like the others, but the class << self made that really weird, so I just went back to the explicit methods that worked...

Comment (stemps) [inline lib/routific.rb]: what happens if you make that an `attr_reader`?

Comment (edwardmccaughan) [inline lib/routific.rb]: scope weirdness- the @logger variable is only connected to the attr_reader some of the time, so it ends up being nil when I try and give it a default value like the other attributes, even though the setter function still works.

I showed it to bodo we couldn't figure it out, so we just went back to the version that worked...

Comment (PragTob) [inline lib/routific.rb]: oh scoping in ruby classes, one of my favorites - if you want we can take a look :)

**Comment by PragTob** [comment]
So the scoping and the documentation is keeping this one from being merged, right? I think we also didn't document our other config options :| might get those in right here or in another PR

**Comment by PragTob** [comment]
Seems like documentation is fixed and we figured out the weird thing about that variable and decided it'd be better to keep as is here. @stemps: agreed and mergable? :)

**Pull request merged**